modules/end-user-guide/partials/proc_using-a-git-provider-access-token.adoc

```diff
@@ -0,0 +1,70 @@
+:parent-context-of-using-a-git-provider-access-token: {context}
+
+[id="using-a-git-provider-access-token_{context}"]
+= Using a Git provider access token
+
+If the administrator of your organization's {prod-short} instance has not configured xref:administration-guide:oauth-for-github-gitlab-or-bitbucket.adoc[OAuth for GitHub, GitLab, or Bitbucket], you can apply your personal access token as a {kubernetes} Secret.
+
+Mounting your access token as a Secret enables the {prod-short} Server to access the remote repository that is cloned during workspace creation, including access to the repository's `/.che` and `/.vscode` folders.
+
+Apply the {kubernetes} Secret in your user {orch-namespace} of the {orch-name} cluster of your organization's {prod-short} instance.
+
+After you have applied the Secret, you can create new workspaces from a private GitHub, GitLab, or Bitbucket-server repository.
+
+TIP: In your user {orch-namespace}, you can create and apply multiple access-token Secrets per a Git provider.
+
+.Prerequisites
+
+* An active `{orch-cli}` session, with administrative permissions, to the {orch-name} cluster. See {orch-cli-link}.
+
+* The link:https://www.gnu.org/software/coreutils/base64[`base64`] command line tools are installed in the operating system you are using.
+
+.Procedure
+
+. Copy your access token and encode it to Base64.
++
+[source,terminal,subs="+quotes,+attributes"]
+----
+$ echo -n '__<your_access_token>__' | base64
+----
+
+. Prepare a new {orch-name} Secret in your user {orch-namespace}.
++
+[source,yaml,subs="+quotes,+attributes,+macros"]
+----
+kind: Secret
+apiVersion: v1
+metadata:
+  name: personal-access-token-__<your_chosen_name_for_this_token>__
+  labels:
+    app.kubernetes.io/component: scm-personal-access-token
+    app.kubernetes.io/part-of: che.eclipse.org
+  annotations:
+    che.eclipse.org/che-userid: _<{prod-id-short}_user_ID>_ <1>
+    che.eclipse.org/scm-personal-access-token-name: _<git_provider_name>_ <2>
+    che.eclipse.org/scm-url: _<Git_provider_endpoint>_ <3>
+    che.eclipse.org/scm-userid: _<Git_provider_user_ID>_ <4>
+    che.eclipse.org/scm-username: _<Git_provider_username>_
+data:
+  token: __<Base64_encoded_access_token>__
+type: Opaque
+----
++
+<1> Your Che user ID. You can retrieve `__<che-endpoint>__/api/user` to get the Che user data.
+<2> The Git provider name: `github` or `gitlab` or `bitbucket-server`.
+<3> The Git provider URL.
+<4> Your Git provider user ID, follow the API documentation to retrieve the user object:
+* GitHub: https://docs.github.com/en/rest/users/users#get-a-user.
+* GitLab: https://docs.gitlab.com/ee/api/users.html#for-normal-users.
+* Bitbucket Server: https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get. Get the `account_id` value.
```

Comment:
```suggestion
<4> Your Git provider user ID, follow the API documentation to retrieve the user object:
* link:https://docs.github.com/en/rest/users/users#get-a-user[GitHub]
* link:https://docs.gitlab.com/ee/api/users.html#for-normal-users[GitLab]
* link:https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get[Bitbucket Server] Get the `account_id` value.
```

Reply:
I need to check this in the preview now.

Reply:
![image](https://user-images.githubusercontent.com/78564304/180790243-a939de38-cbb1-4c4a-b493-a16c520f2575.png)

Reply:
BTW, I can't find `account_id` value in the Bitbucket Server docs (https://developer.atlassian.com/server/bitbucket/rest/v802/api-group-users/#api-users-userslug-get).

Reply:
> List users: For normal users

The idea is to get user by specific `username`, the request example is listed below, so I think it should be named like: `Get a user`.
> I can't find the account_id value in the Bitbucket Server docs

I don't know why the documentation page shows such example, but the real request returns such JSON:
```JSON
{
  "username": "ivinokur",
  "has_2fa_enabled": null,
  "display_name": "ivinokur",
  "account_id": "id-string",
  "links": {
    "hooks": {
      "href": "https://api.bitbucket.org/2.0/workspaces/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D/hooks"
    },
    "self": {
      "href": "https://api.bitbucket.org/2.0/users/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D"
    },
    "repositories": {
      "href": "https://api.bitbucket.org/2.0/repositories/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D"
    },
    "html": {
      "href": "https://bitbucket.org/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D/"
    },
    "avatar": {
      "href": "https://secure.gravatar.com/avatar/f583ff5c1e2669bc67f395b6c8c8c046?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FI-2.png"
    },
    "snippets": {
      "href": "https://api.bitbucket.org/2.0/snippets/%7Be3666240-4b13-4d3e-928f-32d5692c8aa3%7D"
    }
  },
  "nickname": "ivinokur",
  "created_on": "2016-06-29T08:53:43.616762+00:00",
  "is_staff": false,
  "location": null,
  "account_status": "active",
  "type": "user",
  "uuid": "{string}"
}
```

Reply:
Thanks for the clarification.

> The idea is to get user by specific username, the request example is listed below, so I think it should be named like: Get a user.
I use the heading that appears in the linked docs to help the reader find the correct section there. By the way, what should the user look in the GitLab docs? `"id"`?

> I don't know why the documentation page shows such example, but the real request returns such JSON:
Great. Thanks. I updated my suggestion above.

Looking at the linked GitHub docs again, what's the specific parameter that the reader should be looking for? We can add it here too.

Reply:
Here is the updatet version:
![screenshot-0 0 0 0_4000-2022 07 26-12_02_20](https://user-images.githubusercontent.com/7668752/180968050-31a6b85a-e1d6-46e2-92f5-4066c18f60fc.png)

Reply:
Thank you. I'm resolving this conversation.